integration/helpers.go

```diff
@@ -467,22 +467,35 @@ func (i *TeleInstance) StartNode(name string, sshPort int) (*service.TeleportPro
 	tconf.SSH.Enabled = true
 	tconf.SSH.Addr.Addr = net.JoinHostPort(i.Hostname, fmt.Sprintf("%v", sshPort))
 
+	// Create a new Teleport process and add it to the list of nodes that
+	// compose this "cluster".
 	process, err := service.NewTeleport(tconf)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-
 	i.Nodes = append(i.Nodes, process)
 
-	err = process.Start()
+	// Build a list of expected events to wait for before unblocking based off
+	// the configuration passed in.
+	expectedEvents := []string{
+		service.NodeSSHReady,
+	}
+
+	// Start the process and block until the expected events have arrived.
+	_, err = startAndWait(process, expectedEvents)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	// Wait a little bit longer because for some processes, we emit the event
+	// right before the service has started.
+	time.Sleep(250 * time.Millisecond)
```

Comment:
this line defeats the purpose of waiting for event then, I would expect either event or wait, not both

Reply:
@klizhentas I agree, the problem is half of our services have a blocking start and half don't. I can refactor all the services to be non-blocking, but that's probably going to take more time than what we want to spend on 2.4.1.